xpdview/waterfall.py

```diff
@@ -107,8 +106,8 @@ def _adapt_data_list(self, int_data_list):
     def on_plot_hover(self, event):
         """callback to show legend when click on one of curves"""
         line = event.artist
-        name = line.get_label()
-        line.axes.legend([name],handlelength=0,
+        name = line.get_gid()
+        line.axes.legend([name], handlelength=0,
                          handletextpad=0, fancybox=True)
```

Comment:
I would suggest to use ``get_label`` unless special purpose. ``label`` is the natural mpl API